arangod/VocBase/Methods/Collections.h

```diff
@@ -90,7 +90,10 @@ struct Collections {
       bool createWaitsForSyncReplication,             // replication wait flag
       bool enforceReplicationFactor,                  // replication factor flag
       bool isNewDatabase,
-      std::shared_ptr<LogicalCollection>& ret);  // invoke on collection creation
+      std::shared_ptr<LogicalCollection>& ret,
+      bool allowSystem = false,
+      bool isSingleServerEnterpriseCollection = false, // invoke on collection creation
```

Comment:
The comment here is not wrong.
It is related to `ret` which is actually a hidden return value. So please move this up again.
We also seem to get bool input flag creeping here, we should consider using a struct instead, for all people that do not use IDEs to help them to figure out correct name <=> Parameter relation.

Reply:
Ty done. I'm totally on your side. We should refactor this - but I would like to do this (as many other things) in another PR.